client/transport/jsonrpc/src/main/java/io/a2a/client/transport/jsonrpc/JSONRPCTransport.java

```diff
@@ -88,7 +76,7 @@ public JSONRPCTransport(String agentUrl) {
     }
 
     public JSONRPCTransport(AgentCard agentCard) {
-        this(null, agentCard, agentCard.url(), null);
+        this(null, agentCard, agentCard.supportedInterfaces().get(0).url(), null);
```

Comment:
can supportedInterfaces() be empty?

Reply:
According to https://a2a-protocol.org/latest/definitions/, it can be empty:

```
message AgentCard {
  [...]
  // Ordered list of supported interfaces. First entry is preferred.
  repeated AgentInterface supported_interfaces = 19;
}
```

`repeated` means 0 or more

Reply:
@ehsavoie I guess if null/empty we should throw some kind of exception